core/system/src/page_allocator.c

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* This file can be compiled externally to provide the page allocator algorithm
+   for devices NOT supported by uVisor. For this purpose this file is copied as
+   is into uvisor-mbed-lib and compiled by the target build system. */
+#if defined(UVISOR_PRESENT) && (UVISOR_PRESENT == 1)
+#   include <uvisor.h>
+#   include "page_allocator.h"
+#   include "mpu/vmpu_unpriv_access.h"
+#   include "mpu/vmpu.h"
+#   include "halt.h"
+#   include "context.h"
+#else
+#   include "uvisor-lib/uvisor-lib.h"
+#   define DPRINTF(...) {}
+#   define g_active_box 0
+#   define vmpu_is_box_id_valid(...) 0
+#   define HALT_ERROR(id, ...) DPRINTF(__VA_ARGS__)
+static uint32_t vmpu_unpriv_uint32_read(uint32_t addr)
```

Comment:
@niklas-arm please add comment on replacing these two functions with the real unpriv functions headerfile from uVisor.